quickwit/quickwit-metastore/src/metastore/postgres/metastore.rs

```diff
@@ -288,26 +289,29 @@ macro_rules! run_with_tx {
     }};
 }
 
-async fn mutate_index_metadata<E, M: FnOnce(&mut IndexMetadata) -> Result<bool, E>>(
+async fn mutate_index_metadata<E, M>(
     tx: &mut Transaction<'_, Postgres>,
     index_uid: IndexUid,
     mutate_fn: M,
-) -> MetastoreResult<bool>
+) -> MetastoreResult<IndexMetadata>
```

Comment:
- replaced mutable ref in mutation function with move to avoid a systematic clone
- didn't replace the returned mutation state (bool) as it wasn't used anywhere

Reply:
> * replaced mutable ref in mutation function with move to avoid a systematic clone

I don't see `.clone()` calls removed in this file. What am I missing?

> * didn't replace the returned mutation state (bool) as it wasn't used anywhere

👍

Reply:
> I don't see .clone() calls removed in this file. What am I missing?

Actually you are right, the closure can completely use a mutable ref, it doesn't impact the ownership inside `mutate_index_metadata`. My bad!